### PR TITLE
Update requirements

### PR DESCRIPTION
**Pull request by hardbyte**

### CODE REVIEW

**Comment by giorgiop** [comment]
Can you rebase and see if travis is happy after removing python nightly?

**Comment by giorgiop** [comment]
As discussed, it would better to specify requirements with boundaries `=>, <`. 
For travis, we should use `matrix` to see a few combination of python and dependencies versions. 

Let's mention clearly in the README which version with require/support, including python 2 (non)-compatibility.

**Comment by giorgiop** [comment]
Some tests now time out after this
```
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
```

File: .travis.yml [hunk 1]
```diff
@@ -1,48 +1,59 @@
...
+  - pushd mpc-1.0.3
+  - ./configure --prefix=$HOME/.local
+  - make
+  - make install
+  - popd
+  - export CPATH=$HOME/.local/include
+  - export LIBRARY_PATH=$HOME/.local/lib
+  - export LD_LIBRARY_PATH=$HOME/.local/lib
+  - travis_retry pip install gmpy2
+  - if [ "${TESTMODE}" == "full" ]; then pip install coverage; fi
```
Comment (giorgiop) [inline .travis.yml]: Where is this condition triggered? Can we always run with coverage?

File: docs/requirements.txt [hunk 2]
```diff
@@ -1,2 +1,2 @@
-sphinx==1.5.2
-sphinxcontrib-napoleon==0.6.0
+sphinx==1.6.2
```
Comment (giorgiop) [inline docs/requirements.txt]: >= ?

File: .travis.yml [hunk 3]
```diff
@@ -1,48 +1,59 @@
...
+  - travis_retry pip install -U pip setuptools
+  - if [ "${USE_WHEEL}" == "1" ]; then pip install wheel; fi
+  - if [ "${TEST_PYCRYPTO}" == "1" ]; then pip install pycrypto>=2.6.1; fi
+  - travis_retry pip install nose numpy click
+  - python -V
+  - popd
+  - set -o pipefail
 
+script: nosetests -v
```
Comment (hardbyte) [inline .travis.yml]: Yes that would be a great idea. I'll open an issue though

File: docs/requirements.txt (hunk 2 above)
Comment (hardbyte) [inline docs/requirements.txt]: This is less important - just the requirements for building the documentation. I think it is okay to be pinned.

File: .travis.yml (hunk 1 above)
Comment (hardbyte) [inline .travis.yml]: Coverage does slow things down a bit, so I wanted to just run it for python 3.5.

**Comment by giorgiop** [comment]
Can we add test timing with (nose-timer)[https://pypi.python.org/pypi/nose-timer]

**Comment by giorgiop** [comment]
Great, it works. We could also add `--timer-top-n 10` so that we have a nice timing summary at the end.

**Pull request merged**